ui/selectors/assets.ts

```diff
@@ -40,3 +59,152 @@ export function getAssetsMetadata(state: AssetsState) {
+        tokenList.forEach((token: Token) => {
+          const { isNative, address, decimals } = token;
+          const balance =
+            calculateTokenBalance({
+              isNative,
+              chainId,
+              address,
+              decimals,
+              nativeBalances,
+              selectedAccountTokenBalancesAcrossChains,
+            }) || '0';
+
+          const tokenFiatAmount = calculateTokenFiatAmount({
+            token,
+            chainId,
+            balance,
+            marketData,
+            currencyRates,
+          });
+
+          // Respect the "hide zero balance" setting (when true):
+          // - Native tokens should always display with zero balance when on the current network filter.
+          // - Native tokens should not display with zero balance when on all networks filter
+          // - ERC20 tokens with zero balances should respect the setting on both the current and all networks.
+
+          // Respect the "hide zero balance" setting (when false):
+          // - Native tokens should always display with zero balance when on the current network filter.
+          // - Native tokens should always display with zero balance when on all networks filter
+          // - ERC20 tokens always display with zero balance on both the current and all networks filter.
+          if (
+            !hideZeroBalanceTokens ||
+            balance !== '0' ||
+            (token.isNative && isOnCurrentNetwork)
+          ) {
+            tokensWithBalance.push({
+              ...token,
+              balance,
+              tokenFiatAmount,
+              chainId,
+              string: String(balance),
+              primary: '',
+              secondary: 0,
+              title: '',
+            });
+          }
+        });
+      },
+    );
+    return tokensWithBalance;
+  },
+);
+
+export const getMultiChainAssets = createDeepEqualSelector(
+  (_state, selectedAccount) => selectedAccount,
+  getMultichainBalances,
+  getAccountAssets,
+  getAssetsMetadata,
+  getAssetsRates,
+  (_state) => _state.metamask.conversionRates,
+  (
+    selectedAccountAddress,
+    multichainBalances,
+    accountAssets,
+    assetsMetadata,
+    assetRates,
+    multichainCoinRates,
+  ) => {
+    const assetIds = accountAssets?.[selectedAccountAddress.id] || [];
+    const balances = multichainBalances?.[selectedAccountAddress.id];
+    return assetIds.map((assetId: CaipAssetId) => {
+      const [chainId, assetDetails] = assetId.split('/');
+      const isNative = assetDetails.split(':')[0] === 'slip44';
+      const balance = balances?.[assetId] || { amount: '0', unit: '' };
+      const rate = assetRates?.[assetId]?.rate || '0';
+      const nativeRate = multichainCoinRates?.[assetId]?.rate || '0';
+      const balanceInFiat = new BigNumber(balance.amount).times(rate);
+      const nativeBalanceInFiat = new BigNumber(balance.amount).times(
+        nativeRate,
+      );
```

Comment:
Looking at which selectors are used to get `rate` and `nativeRate` it seems like they are coming from the same `conversionRates` state property. Probably something we can de-dupe

Reply:
updated here [7cea61b](https://github.com/MetaMask/metamask-extension/pull/30389/commits/7cea61b7b028720cef3c9419bc25e7c98aab6080)
now the native fiat value we see in asset list should match the one displayed in the main coin-overview component

Reply:
I left another comment about this: https://github.com/MetaMask/metamask-extension/pull/30389#discussion_r1963603244

These other rates should not be used, we should only rely on the `MultichainAssetsRatesController` even in the coin overview, if that means we need to update the coin overview we should do it and not add another patch here.

Reply:
For other reviewers, this is currently being discussed with BTC team 🙏